### DISCUSSION TITLE
@mia-platform/custom-plugin-lib and fastify version compatibility

### DISCUSSION THREAD

**Discussion by rafpor75** [mia-platform/community#530] [Q&A] [answered]
### Description

to fix severe vulnerabilities using "@mia-platform/custom-plugin-lib": "^5.0.0", I updated nodejs from 16.x to 20.x and upgraded @mia-platform/custom-plugin-lib to version 7.0.0
consequently updating fastify to the latest version 4.28.1)
but it seems not to work with a long series of errors of this type:

```
node_modules/@mia-platform/custom-plugin-lib/node_modules/fastify/types/instance.d.ts:275:3 - error TS1109: Expression expected.

275 ): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider>;
```

could you recommend me all the correct versions?

**Comment by davidebianchi** [comment, marked as answer]
Hi @rafpor75! Have you followed the migration guide of the custom-plugin-lib? [Here](https://github.com/mia-platform/custom-plugin-lib/blob/master/docs/migration_guides/10_migrate_from_v6_to_v7.md) the link to the guide.
Furthermore, it is also a fastify major bump migration, have you followed the migration guide of fastify? custom-plugin-lib rely on lc39, which has a reference to the fastify migration guide with some suggestion on what could bring some problems: [migration guide](https://docs.mia-platform.eu/docs/runtime_suite_libraries/lc39/changelog#v700---2023-05-11).
Could be something of this?

**Reply by rafpor75** [reply]
Thank you @davidebianchi  i'll follow the migration guides
